src/lib/mina/actions/offchain-contract.unit-test.ts

```diff
@@ -140,16 +146,145 @@ await testLocal(
+const settleTx = Mina.transaction(sender, async () => {
+  await contractA.settle(proofA);
+  await contractB.settle(proofB);
+});
+await settleTx.sign([sender.key]);
+await settleTx.prove();
+await settleTx.send().wait();
+console.timeEnd('settle');
+
+console.log('Initial supply:');
+console.log(
+  'Contract A total Supply: ',
+  (await contractA.getSupply()).toBigInt()
+);
+console.log(
+  'Contract B total Supply: ',
+  (await contractB.getSupply()).toBigInt()
+);
+assert((await contractA.getSupply()).toBigInt() == 1000n);
+assert((await contractB.getSupply()).toBigInt() == 1500n);
+
+console.log('Initial balances:');
+console.log(
+  'Contract A, Sender: ',
+  (await contractA.offchainState.fields.accounts.get(sender)).value.toBigInt()
+);
+console.log(
+  'Contract B, Sender: ',
+  (await contractB.offchainState.fields.accounts.get(sender)).value.toBigInt()
+);
+assert((await contractA.getBalance(sender)).toBigInt() == 1000n);
+assert((await contractB.getBalance(sender)).toBigInt() == 1500n);
+
+console.time('transfer');
+const transferTx = Mina.transaction(sender, async () => {
+  await contractA.transfer(sender, receiver, UInt64.from(100));
+  await contractB.transfer(sender, receiver, UInt64.from(200));
+});
+await transferTx.sign([sender.key]);
+await transferTx.prove();
+await transferTx.send().wait();
+console.timeEnd('transfer');
+
+console.time('settle');
+const proofA2 = await contractA.offchainState.createSettlementProof();
+const proofB2 = await contractB.offchainState.createSettlementProof();
+const settleTx2 = Mina.transaction(sender, async () => {
+  await contractA.settle(proofA2);
+  await contractB.settle(proofB2);
+});
+await settleTx2.sign([sender.key]);
+await settleTx2.prove();
+await settleTx2.send().wait();
+console.timeEnd('settle');
+
+console.log('Final supply:');
+console.log(
+  'Contract A total Supply: ',
+  (await contractA.getSupply()).toBigInt()
+);
+console.log(
+  'Contract B total Supply: ',
+  (await contractB.getSupply()).toBigInt()
+);
+assert((await contractA.getSupply()).toBigInt() == 1000n);
+assert((await contractB.getSupply()).toBigInt() == 1500n);
+
+console.log('Final balances:');
+console.log(
+  'Contract A, Sender: ',
+  (await contractA.offchainState.fields.accounts.get(sender)).value.toBigInt()
+);
+console.log(
+  'Contract B, Sender: ',
+  (await contractB.offchainState.fields.accounts.get(sender)).value.toBigInt()
+);
+console.log(
+  'Contract A, Receiver: ',
+  (await contractA.offchainState.fields.accounts.get(receiver)).value.toBigInt()
+);
+console.log(
+  'Contract B, Receiver: ',
+  (await contractB.offchainState.fields.accounts.get(receiver)).value.toBigInt()
+);
+assert((await contractA.getBalance(sender)).toBigInt() == 900n);
+assert((await contractB.getBalance(sender)).toBigInt() == 1300n);
+assert((await contractA.getBalance(receiver)).toBigInt() == 100n);
+assert((await contractB.getBalance(receiver)).toBigInt() == 200n);
```

Comment:
imo it would be good to add a test where the state of both contracts actually diverge more aggressively (eg contract A doesn't settle for 10 actions, whereas contract B settles every action)

Reply:
Ok, I will add this kind of test, rm the extra type I added, noted in Boray's comment, and add to the changelog.  Anything else you want updated?

Reply:
I need to get a bit familiar with the changes first, will do another pass